src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {...currentState, ...action.extraData,};
+        break;
+
+      case 'removeProperties':
+        const { keysToRemove } = action;
+        const newState = {};
+
+        for (const key in currentState) {
+          if (!keysToRemove.includes(key)) {
+            newState[key] = currentState[key];
+          }
+        }
+
+        currentState = newState;
```

Comment:
this code could be simplified to work only with `keysToRemove`
```suggestion
        for (const key of keysToRemove) {
          delete currentState[key];
        }
```